control-plane/csi-driver/src/bin/node/fsfreeze/mod.rs

```diff
@@ -0,0 +1,47 @@
+pub(crate) use bin::FsFreezeOpt;
+use csi_driver::limiter::VolumeOpGuard;
+use tokio::process::Command;
+use tonic::Status;
+use tracing::debug;
+
+pub(crate) mod bin;
+
+// Default csi-node binary name.
+const CSI_NODE_BINARY: &str = "csi-node";
+// Volume name arg for fs freeze and unfreeze subcommand.
+const FREEZE_VOLUME_ID_ARG_NAME: &str = "--volume-id";
+
+/// Creates a child process that calls up the csi-node binary to execute FIFREEZE and FITHAW using
+/// the volume id.
+pub async fn fsfreeze(volume_id: &str, freeze_op: FsFreezeOpt) -> Result<(), Status> {
+    let volume_id = volume_id.to_string();
+    let binary_name = std::env::args().next().unwrap_or(CSI_NODE_BINARY.into());
+    tokio::spawn(async move {
+        debug!(volume_id = volume_id, "{freeze_op}");
+        let _guard = VolumeOpGuard::new_str(&volume_id)?;
+        match Command::new(binary_name)
+            .arg(freeze_op.as_ref())
+            .arg(FREEZE_VOLUME_ID_ARG_NAME)
+            .arg(&volume_id)
+            .output()
+            .await
+        {
+            Ok(output) => {
+                if output.status.success() {
+                    Ok(())
+                } else {
+                    Err(Status::internal(format!(
+                        "{freeze_op} for {volume_id}, failed with {}, error: {}",
```

Comment:
I might be reading this wrong, but this means we don't get any meaningful error back?

Reply:
We do get the error back, the error contains the Errno

Reply:
Now since this is an error from the stderr from the Command output we cannot have very nice mapping from Stderr string to tonic status. That's why most of it is internal or aborted.

For example below gets returned to the client when volume is not found/ preflight check fails.

`fs-freeze for ec4e66fd-3b33-4439-b504-d49aba53da26, failed with exit status: 1, error: Error: Cannot find volume: volume ID: ec4e66fd-3b33-4439-b504-d49aba53da26`

`fs-freeze for ec4e66fd-3b33-4439-b504-d49aba53da26, failed with exit status: 1, error: Error: fsfreeze failed: volume ID: ec4e66fd-3b33-4439-b504-d49aba53da26, EHOSTUNREACH: No route to host`

Reply:
Stderr? But we get the actual error back via status right?

Reply:
No I mean in this method where we call the command we get the error from the stderr, we return a tonic status to grpc client. But since this method returns in stderr we don't have a very good way of mapping the stderr to specific tonic status.

Reply:
Done.